File: Objective-C/Reference App/4. Search for a Location/SearchForLocationViewController.m
//
//  SearchForLocationViewController.m
//  Reference App
//
//  Created by David Deller on 5/3/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

#import "SearchForLocationViewController.h"

@interface SearchForLocationViewController ()

@end

@implementation SearchForLocationViewController

- (void)handleInput:(NSString *)input {
    [self searchUsingAddress:input];
}

- (void)searchUsingAddress:(nonnull NSString *)address {
    CLLocationCoordinate2D currentLocation = CLLocationCoordinate2DMake(34.101558, -118.340944); // Grauman's Chinese Theatre
    
    TGSearchRequest *request = [TGSearchRequest.alloc initWithSingleLine:address location:currentLocation searchExtent:nil suggestionKey:nil];
    
    [TGFindService searchWithRequest:request completionHandler:^(TGSearchResponse * _Nonnull response) {
        NSArray<TGSearchResult *> *results = response.results;
        // do something with results
        
        [self handleOutput:results];
    }];
}


@end

File: Swift 4/Reference App/302. Get detailed location information from lat long pair/SearchForLocationViewController.swift
//
//  SearchForLocationViewController.swift
//  Reference App
//
//  Created by David Deller on 5/1/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

import UIKit
import TallyGoKit

class SearchForLocationViewController: InputOutputViewController {
    
    override func handleInput(_ input: String) {
        search(using: input)
    }

    func search(using address: String) {
        let currentLocation = CLLocationCoordinate2D(latitude: 34.101558, longitude: -118.340944) // Grauman's Chinese Theatre
        
        let request = TGSearchRequest(singleLine: address, location: currentLocation, searchExtent: nil, suggestionKey: nil)
        
        TGFindService.search(with: request) { (response) in
            let results = response.results
            // do something with results
            
            self.handleOutput(results)
        }
    }


}
